lib/utils/navigation_controller.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:injectable/injectable.dart';
+import 'package:kayla_flutter_ic/di/di.dart';
+
+final navigationController = getIt.get<NavigationController>();
+
+@singleton
+class NavigationController {
+  void push(BuildContext context, Widget widget) {
+    Route route =
+        MaterialPageRoute(builder: (context) => ProviderScope(child: widget));
```

Comment:
we just need to pass `ProviderScope` once in `main.dart` : `runApp(ProviderScope(child: MyApp()));`. Then we can get rid of `ProviderScope` here

Reply:
Error when removing the ProviderScope.

<img width=300 src="https://user-images.githubusercontent.com/23162627/218951004-801fa9a8-94b3-470b-ac89-cc44e8d6a900.gif">